x-pack/test/functional_enterprise_search/services/app_search_service.ts

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { FtrProviderContext } from '../ftr_provider_context';
+
+const ENTERPRISE_SEARCH_USER = 'enterprise_search';
+const ENTERPRISE_SEARCH_PASSWORD = 'changeme';
+import {
+  createEngine,
+  createMetaEngine,
+  indexData,
+  waitForIndexedDocs,
+  destroyEngine,
+  IEngine,
+} from './app_search_client';
+
+export interface IUser {
+  user: string;
+  password: string;
+}
+export { IEngine };
+
+export class AppSearchService {
```

Comment:
I'm just curious - I feel like I've almost grasped this but would also love a 'human-readable' explanation so to speak. What differentiates the concept of the App Search **Client** from the App Search **Service**? I get that the client manages the nitty-gritty http calls - is the Service just a thin and easier-to-read wrapper on top of that?

Reply:
Services are a first class concept in Kibana: https://www.elastic.co/guide/en/kibana/current/development-functional-tests.html#_custom_services. Having an App Search service to encapsulate all of the setup / interaction with App Search seemed ideal.

Splitting out the low level API interactions to a client just cleans up the Service.

I'm primarily using the Service like a Factory in Rails https://github.com/elastic/ent-search/blob/master/spec/factories/engines.rb, creating different types of test fixtures... createComplexEngine, createSimpleEngine, createPokemonEngine, etc.

Reply:
Ahhh, thanks, that helped clarify that for me!